src/invariant/test/OrderBookIsNotCrossedTests.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "invariant/InvariantDoesNotHold.h"
+#include "invariant/InvariantManager.h"
+#include "invariant/OrderBookIsNotCrossed.h"
+#include "invariant/test/InvariantTestUtils.h"
+#include "ledger/LedgerTxn.h"
+#include "test/TestUtils.h"
+#include "test/TxTests.h"
+#include "test/test.h"
+
+using namespace stellar;
+using namespace stellar::InvariantTestUtils;
+
+#define SHOULD_THROW false
+#define SHOULD_PASS true
+
+// XNOR's truth table is as follows:
+// | a | b | a XNOR b |
+// |---|---|----------|
+// | 0 | 0 |     1    |
+// | 0 | 1 |     0    |
+// | 1 | 0 |     0    |
+// | 1 | 1 |     1    |
+bool
+XNOR(bool a, bool b)
+{
+    return (a && b) || (!a && !b);
+}
+
+void
+genApplyCheckCreateOffer(Asset& ask, Asset& bid, int64 amount, Price price,
+                         Application& app, bool shouldPass)
+{
+    auto offer = generateOffer(ask, bid, amount, price);
+    auto op = txtest::manageOffer(offer.data.offer().offerID, ask, bid, price,
+                                  amount);
+    std::vector<LedgerEntry> current{offer};
+    auto updates = makeUpdateList(current, nullptr);
+    REQUIRE(XNOR(store(app, updates, nullptr, nullptr, &op), shouldPass));
+}
+
+void
+genApplyCheckModifyOffer(Asset& ask, Asset& bid, int64 amount, Price price,
```

Comment:
Passing around the operations is kind of clunky. The invariant only depends on the assets in the operation, so you should be able to prepare them in these functions.